src/puppetlabs/puppetdb/query.clj

```diff
@@ -140,13 +140,22 @@
   function basically just checks that the operator is known, and then
   dispatches to the function implementing it."
   [ops [op & args :as term]]
-  (when-not (sequential? term)
-    (throw (IllegalArgumentException. (format "%s is not well-formed: queries must be an array" (vec term)))))
-  (when-not op
-    (throw (IllegalArgumentException. (format "%s is not well-formed: queries must contain at least one operator" (vec term)))))
+
+  (cond
+
+    (empty? term)
+    {:where nil :params nil}
+    (not (sequential? term))
```

Comment:
The above call to `empty?` will already throw if term can't be turned into a seq.